src/actions.ts

```diff
@@ -92,6 +92,9 @@ export async function executeActions(
         },
       );
     } else if (actionSpec._ === "command") {
+      const hookConnectionString = actionSpec.root
+        ? makeRootDatabaseConnectionString(parsedSettings, databaseName)
+        : connectionString;
```

Comment:
Since you've added `root` to the `ActionSpecBase` let's move this to above the `if` statement so all the action types can use it easily.